DashAI/back/api/api_v1/endpoints/plugins.py

```diff
@@ -253,14 +255,20 @@ async def update_plugin(
         try:
             plugin = db.get(Plugin, plugin_id)
             plugin_name = plugin.name
-            install_plugin_from_pypi(plugin_name)
-            available_plugins: List[type] = get_available_plugins()
-            register_new_plugins(component_registry, available_plugins)
             if not plugin:
                 raise HTTPException(
                     status_code=status.HTTP_404_NOT_FOUND,
                     detail="Plugin not found",
                 )
+            if params.new_status == PluginStatus.INSTALLED:
+                installed_components = install_plugin(plugin_name)
+                register_plugin_components(installed_components, component_registry)
+            elif (
+                plugin.status == PluginStatus.INSTALLED
+                and params.new_status == PluginStatus.REGISTERED
+            ):
+                uninstalled_components = uninstall_plugin(plugin_name)
+                unregister_plugin_components(uninstalled_components, component_registry)
```

Comment:
I don't think it happens at the moment, but it could happen that there are plugins that are dependent on others?

And the other question also related to uninstallation: When you uninstall the plugin, you only uninstall the plugin package, but not its dependencies right. I think that by default pip does that, but better to be clear about it

Reply:
Uninstalling a plugin doesn't uninstall the dependencies.